File: frontend/src/types/cuaderno.ts
export interface NotaCuaderno {
  id: number;
  user_id: number;
  potrero_id: number | null;
  texto: string;
  created_at: string;
}

export interface NotaCreate {
  texto: string;
  potrero_id?: number | null;
}

export interface TareaCuaderno {
  id: number;
  user_id: number;
  potrero_id: number | null;
  texto: string;
  fecha_planificada: string | null;
  completada: boolean;
  completed_at: string | null;
  notificar_dias_antes: number | null;
  created_at: string;
}

export interface TareaCreate {
  texto: string;
  fecha_planificada?: string | null;
  potrero_id?: number | null;
  notificar_dias_antes?: number | null;
}

File: backend/app/models/cuaderno.py
from datetime import date, datetime
from typing import Optional

from sqlalchemy import Boolean, Date, DateTime, ForeignKey, Integer, Text
from sqlalchemy.orm import Mapped, mapped_column
from sqlalchemy.sql import func

from app.database import Base


class NotaCuaderno(Base):
    __tablename__ = "notas_cuaderno"

    id: Mapped[int] = mapped_column(primary_key=True, index=True)
    user_id: Mapped[int] = mapped_column(
        Integer, ForeignKey("users.id", ondelete="CASCADE"), nullable=False, index=True
    )
    potrero_id: Mapped[Optional[int]] = mapped_column(
        Integer, ForeignKey("potreros.id", ondelete="SET NULL"), nullable=True, index=True
    )
    texto: Mapped[str] = mapped_column(Text, nullable=False)
    created_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True), server_default=func.now(), nullable=False
    )


class TareaCuaderno(Base):
    __tablename__ = "tareas_cuaderno"

    id: Mapped[int] = mapped_column(primary_key=True, index=True)
    user_id: Mapped[int] = mapped_column(
        Integer, ForeignKey("users.id", ondelete="CASCADE"), nullable=False, index=True
    )
    potrero_id: Mapped[Optional[int]] = mapped_column(
        Integer, ForeignKey("potreros.id", ondelete="SET NULL"), nullable=True, index=True
    )
    texto: Mapped[str] = mapped_column(Text, nullable=False)
    fecha_planificada: Mapped[Optional[date]] = mapped_column(Date, nullable=True)
    completada: Mapped[bool] = mapped_column(
        Boolean, nullable=False, server_default="false"
    )
    completed_at: Mapped[Optional[datetime]] = mapped_column(
        DateTime(timezone=True), nullable=True
    )
    notificar_dias_antes: Mapped[Optional[int]] = mapped_column(
        Integer, nullable=True, server_default="1"
    )
    created_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True), server_default=func.now(), nullable=False
    )
